sympy/solvers/ode/ode.py

```diff
@@ -708,7 +710,8 @@ def _helper_simplify(eq, hint, match, simplify=True, ics=None, **kwargs):
             rv = _handle_Integral(exprs, func, hint)
 
     if isinstance(rv, list):
-        rv = _remove_redundant_solutions(eq, rv, order, func.args[0])
+        if hint != "2nd_nonlinear_autonomous_conserved" and hint != "2nd_nonlinear_autonomous_conserved_Integral" or simplify:
+            rv = _remove_redundant_solutions(eq, rv, order, func.args[0])
```

Comment:
We shouldn't check for specific hints here.

I think just checking for simplify should be fine if that means we can test the solver. The real fix is to make it so that `_remove_redundant_solutions` only handles simple cases e.g. those that are linear in the constants.

Reply:
Okay, will remove these